### PR TITLE
[refactor](stats) Use id instead name in analysis info

### PR DESCRIPTION
**Pull request by Kikyou1997**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java [hunk 1]
```diff
@@ -2663,9 +2665,16 @@ private void handleShowAnalyze() {
...
-            row.add(analysisInfo.catalogName);
-            row.add(analysisInfo.dbName);
-            row.add(analysisInfo.tblName);
+            CatalogIf<? extends DatabaseIf<? extends TableIf>> c = StatisticsUtil.findCatalog(analysisInfo.catalogId);
+            row.add(c.getName());
+            Optional<? extends DatabaseIf<? extends TableIf>> databaseIf = c.getDb(analysisInfo.dbId);
+            row.add(databaseIf.isPresent() ? databaseIf.get().getFullName() : "DB may get deleted");
+            if (databaseIf.isPresent()) {
+                Optional<? extends TableIf> table = databaseIf.get().getTable(analysisInfo.tblId);
+                row.add(table.isPresent() ? table.get().getName() : "Table may get deleted");
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java:2674]: if db is deleted, how to handle this error?
why there is no lock?

File: fe/fe-core/src/main/java/org/apache/doris/analysis/AnalyzeTblStmt.java [hunk 2]
```diff
@@ -89,6 +89,8 @@ public class AnalyzeTblStmt extends AnalyzeStmt {
     private PartitionNames partitionNames;
     private boolean isAllColumns;
 
+    private long catalogId;
+
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/AnalyzeTblStmt.java:93]: should add this var after comment in `L94`

File: fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisManager.java [hunk 3]
```diff
@@ -398,9 +401,16 @@ private void sendJobId(List<AnalysisInfo> analysisInfos, boolean proxy) {
...
-            row.add(analysisInfo.tblName);
+            CatalogIf<? extends DatabaseIf<? extends TableIf>> c = StatisticsUtil.findCatalog(analysisInfo.catalogId);
+            row.add(c.getName());
+            Optional<? extends DatabaseIf<? extends TableIf>> databaseIf = c.getDb(analysisInfo.dbId);
+            row.add(databaseIf.isPresent() ? databaseIf.get().getFullName() : "DB may get deleted");
+            if (databaseIf.isPresent()) {
+                Optional<? extends TableIf> table = databaseIf.get().getTable(analysisInfo.tblId);
+                row.add(table.isPresent() ? table.get().getName() : "Table may get deleted");
+            } else {
+                row.add("DB may get deleted");
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisManager.java:412]: not exists

**Review by morrySnow — DISMISSED** [review]

**Comment by Kikyou1997** [comment]
run buildall

**Review by englefly — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Pull request merged**